components/core_metrics/core_metrics_service.cc

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/core_metrics/core_metrics_service.h"
+
+#include "base/bind.h"
+#include "base/logging.h"
+#include "base/time/time.h"
+#include "brave/components/core_metrics/pref_names.h"
+#include "brave/components/p3a_utils/bucket.h"
+#include "brave/components/time_period_storage/weekly_storage.h"
+#include "components/history/core/browser/history_service.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+
+namespace core_metrics {
+
+namespace {
+
+constexpr const int kPagesLoadedBuckets[] = {0, 10, 50, 100, 500, 1000};
+constexpr const int kDomainsLoadedBuckets[] = {0, 4, 10, 30, 50, 100};
+
+constexpr base::TimeDelta kPagesLoadedReportIntervalMinutes = base::Minutes(30);
+constexpr base::TimeDelta kDomainsLoadedReportIntervalMinutes =
+    base::Minutes(30);
+
+}  // namespace
+
+CoreMetricsService::CoreMetricsService(PrefService* local_state,
+                                       history::HistoryService* history_service)
+    : local_state_(local_state), history_service_(history_service) {
+  DCHECK(local_state);
+  DCHECK(history_service);
+
+  pages_loaded_report_timer_.Start(FROM_HERE, kPagesLoadedReportIntervalMinutes,
+                                   this,
+                                   &CoreMetricsService::ReportPagesLoaded);
+  domains_loaded_report_timer_.Start(FROM_HERE,
+                                     kDomainsLoadedReportIntervalMinutes, this,
+                                     &CoreMetricsService::ReportDomainsLoaded);
+  ReportPagesLoaded();
+  ReportDomainsLoaded();
+}
+
+CoreMetricsService::~CoreMetricsService() = default;
+
+void CoreMetricsService::RegisterPrefs(PrefRegistrySimple* registry) {
+  registry->RegisterListPref(kCoreMetricsPagesLoadedCount);
+}
+
+void CoreMetricsService::IncrementPagesLoadedCount() {
+  VLOG(2) << "CoreMetricsService: increment page load count";
+  if (pages_loaded_storage_ == nullptr) {
+    pages_loaded_storage_ = std::make_unique<WeeklyStorage>(
+        local_state_, kCoreMetricsPagesLoadedCount);
+  }
+  pages_loaded_storage_->AddDelta(1);
+}
+
+void CoreMetricsService::ReportDomainsLoaded() {
+  history_service_->GetDomainDiversity(
+      base::Time::Now(), 1, history::DomainMetricType::kEnableLast7DayMetric,
```

Comment:
What `1` stands for here? `number_of_days_to_report`?

Reply:
yes, will add comment before param